docs/src/proposals/fee_transaction_priority.md

```diff
@@ -0,0 +1,210 @@
+processed that contends on an account `A`, then `T2` should not be scheduled if it would grab
+any account locks needed by `T1`. This prevents lower fee transactions like `T2` from starving
+higher paying transactions like `T1`.
+
+
+### Transaction Pipeline
+
+Pipeline:
+1. Sigverify
+2. Scheduler
+3. BankingStage threads
+
+Transactions from sigverify are connected via a channel to the scheduler. The scheduler maintains
+`N` bi-directional channels with the `N` BankingStage threads, implemented as a pair of two
+unidirectional channels.
+
+The scheduler's job is to run an algorithm in which it determines how to schedule transactions
+received from sigverify to the `N` BankingStage threads. A transaction is scheduled to be executed
+on a particular BankingStage thread by sending that transaction to the thread via its associated
+channel.
+
+Once a BankingStage thread finishes processing a transaction `T` , it sends the `T` back
+to the scheduler via the same channel to signal of completion.
+
+### Scheduler Implementation
+
+The scheduler is the most complex piece of the above pipeline, its implementation is made up of a
+few pieces. Note for now, all these pieces are maintained by the single scheduler thread to avoid
+locking complexity.
+
+#### Components of the `Scheduler`:
+
+1. `default_transaction_queue` - A max-heap `BinaryHeap<Transaction>` that tracks all pending transactions.
+The priority in the heap is the additional fee of the transaction. Transactions are added to this queue
+from sigverify before the leader slot begins.
+2. `all_transaction_queues` - A `VecDeque<BinaryHeap<Transaction>>` that tracks all pending queues of work.
+Some pending queues have higher priority than others (as will be explained later in the `Handling Completion Signals from BankingStage Threads` section below). The list is ordered in priority from highest to lowest priority. On
+initialization, `all_transaction_queues[0] = default_transaction_queue`.
+3. `locked_accounts` - A `HashMap<LockedPubkey, usize>` that tracks the set of accounts needed to execute the
+current set of transactions scheduled/sent to banking threads. Accounts are added to this set
+*before* being sent to BankingStage threads. The `usize` represents a refcount, because multiple read
+accounts could be grabbed.`LockedPubkey` is defined as:
+```
+enum LockedPubkey {
+    Read(Pubkey),
+    Write(Pubkey),
+}
+```
+4. `blocked_transactions` -  A `HashMap<Signature, Rc<BlockedTransactionsQueue>>` keyed by
+transaction signature, and maps to a `BlockedTransactionsQueue` defined as:
+```
+/// Represents a heap of transactions that cannot be scheduled because they
+/// would take locks on accounts needed by a higher paying transaction
+struct BlockedTransactionsQueue {
+    // The higher priority transactin blocking all the other transactions in
+    // `blocked_transactions` below
+    highest_priority_blocked_transaction: Transaction,
```

Comment:
I had it not part of the heap.  Since this transaction would be referenced/checked a lot it made sense to me to clearly delineate it from the other transactions.